packages/api-client/src/store/active-entities.ts

```diff
@@ -0,0 +1,189 @@
+import { flattenEnvVars } from '@/libs/string-template'
+import { PathId } from '@/router'
+import type { Environment } from '@scalar/oas-utils/entities/environment'
+import type {
+  Collection,
+  Request,
+  RequestExample,
+  Server,
+} from '@scalar/oas-utils/entities/spec'
+import type { Workspace } from '@scalar/oas-utils/entities/workspace'
+import { type InjectionKey, computed, inject } from 'vue'
+import type { Router } from 'vue-router'
+
+import { getRouterParams } from './router-params'
+
+type CreateActiveEntitiesStoreParams = {
+  router: Router
+  collections: Record<string, Collection>
+  environments: Record<string, Environment>
+  requestExamples: Record<string, RequestExample>
+  requests: Record<string, Request>
+  servers: Record<string, Server>
+  workspaces: Record<string, Workspace>
+}
+
+/**
+ * Create the active entities store
+ *
+ * We need the factory function to pass the router instance
+ */
+export const createActiveEntitiesStore = ({
+  collections,
+  environments,
+  requestExamples,
+  requests,
+  router,
+  servers,
+  workspaces,
+}: CreateActiveEntitiesStoreParams) => {
+  /** Gives the required UID usually per route */
+  const activeRouterParams = computed(getRouterParams(router))
+
+  /** The currently selected workspace OR the first one */
+  const activeWorkspace = computed(() => {
+    const workspace =
+      workspaces[activeRouterParams.value[PathId.Workspace]] ??
+      workspaces[Object.keys(workspaces)[0]]
+
+    return workspace
+  })
+
+  /** Ordered list of the active workspace's collections with drafts last */
+  const activeWorkspaceCollections = computed(() =>
+    activeWorkspace.value?.collections
+      .map((uid) => collections[uid])
+      .sort((a, b) => {
+        if (a.info?.title === 'Drafts') return 1
+        if (b.info?.title === 'Drafts') return -1
+        return 0
+      }),
+  )
+
+  /** Simplified list of servers in the workspace for displaying */
+  const activeWorkspaceServers = computed(() =>
+    activeWorkspaceCollections.value?.flatMap((collection) =>
+      collection.servers.map((uid) => servers[uid]),
+    ),
+  )
+
+  /** Simplified list of requests in the workspace for displaying */
+  const activeWorkspaceRequests = computed(() =>
+    activeWorkspaceCollections.value?.flatMap(
+      (collection) => collection.requests,
+    ),
+  )
+
+  /** TODO: need to merge collection environments into the global space */
```

Comment:
No merging into the global space. We should probably be consuming the workspace collection and merging it here with the collection env and then exporting the `activeEnvVars` object

Reply:
ya these comments are just copy pasted from the other store. I can just delete it